### ISSUE TITLE
Magnet URL blocked in Tor Private window

### ISSUE THREAD

**Issue by haplo** [brave/brave-browser#26477]
## Description 
Magnet links open fine in a Private window, but Brave blocks them in Tor Private windows.

## Steps to Reproduce

   1. Open a Private window with Tor
   2. Navigate to https://archlinux.org/download/ 
   3. Click on the magnet link in the BitTorrent download section

## Actual result:

Page blocked by Brave, `ERR_BLOCKED_BY_CLIENT` error.

![image](https://user-images.githubusercontent.com/71658/199483146-62143d78-9b54-452c-8f96-431bdcecefd8.png)

## Expected result:

I would like for the Magnet link to open normally with an external BitTorrent application (I don't use Webtorrent):

![image](https://user-images.githubusercontent.com/71658/199483280-49433455-ad50-4134-91b5-388bc74efb98.png)

However I understand that for some users opening a Magnet link, maybe silently, could harm the anonimity provided by the Tor private window. I think ideally Brave would display a warning page when opening a magnet link in a Tor private window. If this is the adopted solution there should be a "don't ask me again" checkbox.

## Reproduces how often: 
Easily.

## Brave version (brave://version info)
```
Brave | 1.45.116 Chromium: 107.0.5304.91 (Official Build) (64-bit)
-- | --
Revision | 3d5948960d62418160796d5831a4d2d7d6c90fa8-refs/branch-heads/5304@{#1097}
OS | Linux
JavaScript | V8 10.7.193.18
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36
Command Line | /opt/brave.com/brave/brave --enable-crashpad --disable-domain-reliability --enable-dom-distiller --component-updater=url-source=https://go-updater.brave.com/extensions --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --variations-insecure-server-url=https://variations.brave.com/seed --flag-switches-begin --flag-switches-end --desktop-startup-id=tuxedo;1667389093;892720;2798_TIME257847135
Executable Path | /opt/brave.com/brave/brave
Profile Path | /home/fidel/.config/BraveSoftware/Brave-Browser/Default
Active Variations | AdRewardsStudy:NextPaymentDayBraveAdblockCookieListOptInReleaseStudy:EnabledBraveAds.AdServingStudy:DefaultAdNotificationsPerHour=10/MaximumAdNotificationsPerDay=100/MaximumInlineContentAdsPerHour=12/MaximumInlineContentAdsPerDay=60/AdServingVersion=2BraveAds.UserActivityStudy:Triggers=NTPSI/TimeWindow=15m/Threshold=0.0/IdleTimeThreshold=5sBraveDebounceStudy:EnabledBraveRewardsWebUiPanelStudy:EnabledBraveScreenFingerprintingBlockerStudy:DefaultConversionsStudy:NewDefaultPatternDefault1pBlockingStudy:DisabledDisableReduceLanguage:EnabledEphemeralStorageStudy:EnabledFirstPartyEphemeralDomainBlockStudy:EnabledForceDisableBlinkFeatures:EnabledPartitionBlinkMemoryCacheStudy:EnabledPartitionConnectionsByNetworkIsolationKeyStudy:EnabledPermissionLifetimeReleaseStudy:EnabledSpeedreaderReleaseStudy:Enabled
```

## Version/Channel Information:
Reproduces on Brave installed via Ubuntu Snap with edge channel.

## Other Additional Information:
- Does the issue resolve itself when disabling Brave Shields? 
  - Shields are already disabled in magnet links.
- Does the issue resolve itself when disabling Brave Rewards? 
  - No.
- Is the issue reproducible on the latest version of Chrome? 
  - No, seems to be a problem exclusively with Tor private windows in Brave.

**Comment by diracdeltas** [comment]
you're right that this for now is the intended behavior because the privacy (or lack thereof) of the torrent protocol goes against the usual privacy guarantees of tor browsing. for now i think you can workaround this by simply copy pasting the magnet URL into a non-private window.

**Comment by haplo** [comment]
> you're right that this for now is the intended behavior because the privacy (or lack thereof) of the torrent protocol goes against the usual privacy guarantees of tor browsing. for now i think you can workaround this by simply copy pasting the magnet URL into a non-private window.

Thank you for replying! Yes, copy-pasting is the obvious way out, but it's cumbersome if there are multiple torrents you want to go through. A way to disable that behavior would be much appreciated, even if it's an obscure setting that most users won't care about.

**Comment by Tynam** [comment]
> Yes, copy-pasting is the obvious way out, but it's cumbersome if there are multiple torrents you want to go through.

Very much seconded.  I have carpal tunnel; every click is a tiny bit of extra friction.  The difference between "click link" and menu-click/copy link/tab out to app/paste link is huge.  Having an override option somewhere would be extremely helpful, even if most users don't ever need it.

If devs don't want to open that security risk, even having a good keyboard/click shortcut for "copy link" would go a long way.

**Comment by dragonman225** [comment]
I ran into this issue myself, and I think it would be clearer to tell users to open torrent or magnet links outside of private windows, rather than just saying “this page has been blocked.”

When I saw that message, my first instinct was to try disabling the torrent feature so I could get the torrent file and download the content with another software—completely missing the benefit of the built-in torrent support.

Related: #9260, #3098

**Comment by haplo** [comment]
> Closing as duplicate of [#3098](https://github.com/brave/brave-browser/issues/3098)

I don't think this issue is a duplicate, even without webtorrent I think the handling of magnet links could be improved. See the proposal in my original message for an example.

Personally I would just be happy with some obscure hidden setting that we advanced users could set so magnet links are handled like in regular browser sessions.

And as others have pointed out in this thread, the error message that is displayed could be better.
